lib/rex/post/meterpreter/extensions/priv/priv.rb

```diff
@@ -95,6 +95,17 @@ def sam_hashes
     }
   end
 
+  def ntds_parse(filepath)
+    request = Packet.create_request( 'priv_ntds_parse' )
+    request.add_tlv( TLV_TYPE_NTDS_PATH, filepath)
+    response = client.send_request( request, 90 )
+    channel_id = response.get_tlv_value(TLV_TYPE_CHANNEL_ID)
+    if channel_id.nil?
+      raise Exception, "We did not get a channel back!"
+    end
+    channel = Rex::Post::Meterpreter::Channels::Pool.new(client, channel_id, "priv_ntds", CHANNEL_FLAG_SYNCHRONOUS)
```

Comment:
You assign to the local `channel`.  This is either an unread local, in which case `channel =` can be removed, or you meant to do `self.channel =` if `channel` is an attribute.